File: RandomPancakeStackDiv2/solve/python/RandomPancakeStackDiv2Solver.py
#! /usr/bin/env python
import os, sys
import RandomPancakeStackDiv2

def init():
    default_path = os.path.join(os.getenv("HOME"), ".gettc")
    gettc_home = os.path.abspath(os.getenv("GETTC_HOME", default_path))
    include_dir = os.path.join(gettc_home, "include/python")
    sys.path.append(include_dir)

def main():
    import topcoder as tc
    with open(sys.argv[1], "r") as fi:
        input = fi.read()
        reader = tc.Reader(input)
    
        d = reader.next("int[]")

    result = RandomPancakeStackDiv2.expectedDeliciousness(d)
    with open(sys.argv[2], "w") as fo:
        fo.write(tc.write(result, "double"))

if __name__ == "__main__":
    init()
    main()

File: RandomPancakeStackDiv2/solve/ruby/RandomPancakeStackDiv2Solver.rb
#! /usr/bin/env ruby

require_relative "RandomPancakeStackDiv2"

require "gettc/types"
include Gettc

def init()
    gettc_home = File.join(ENV["HOME"], ".gettc")
    if ENV.has_key?("GETTC_HOME")
        gettc_home = ENV["GETTC_HOME"]
    end 
    gettc_home = File.expand_path(gettc_home)
    $LOAD_PATH << File.join(gettc_home, "include", "ruby")
    require "topcoder"
    include TopCoder
end

def main()
    reader = Reader.new(IO.read(ARGV[0]))
    d = reader.next(TArray.new(TInt))

    result = RandomPancakeStackDiv2.new().expectedDeliciousness(d)
    IO.write(ARGV[1], Writer.new().next(result, TDouble).to_s())
end
init()
main()
